src/api.c

```diff
@@ -4266,11 +4269,23 @@ static struct cgroup_rule *cgroup_find_matching_rule_uid_gid(uid_t uid, gid_t gi
 				continue;
 			}
 
-			/* If UID is a member of group, we matched. */
-			for (i = 0; grp->gr_mem[i]; i++) {
-				if (!(strcmp(usr->pw_name, grp->gr_mem[i])))
-					return rule;
+			cgroup_dbg("User name: %s UID: %d\nGroup name: %s GID: %d\n", usr->pw_name,
+					uid, grp->gr_name, grp->gr_gid);
+			if (grp->gr_mem[0] != NULL) {
+				cgroup_dbg("Group member(s):\n");
+				/* If UID is a member of group, we matched. */
+				for (i = 0; grp->gr_mem[i]; i++) {
+					cgroup_dbg("  %s\n", grp->gr_mem[i]);
+					if (!(strcmp(usr->pw_name, grp->gr_mem[i]))) {
+						match_found = true;
+						if (!cgroup_is_dbg && match_found)
+							break;
```

Comment:
The above `if` condition, can be simplified with `if (!cgroup_is_dbg)` as `match_found` is always `true`